### PR TITLE
Allow showing downvotes separately or hiding scores

### LINKED ISSUE #135: Allow hiding scores
Lemmy has an option that removes all counts of votes from posts and comments. I'd like to see a toggle in wefwef for this too. It would hide counts for posts, comments and profile pages as well. 

Related: https://lemmy.world/post/911284

### LINKED ISSUE #170: [Feature request] Show downvote counter
For parity with main Lemmy site

### PR DESCRIPTION
**Pull request by rsammelson**
Fixes #170 and fixes #135. When the setting is in the default mode, it should work the same as before.

upvote<br />
downvote<br />
no votes<br />
selector

### CODE REVIEW

**Comment by rsammelson** [comment]
I accidentally ended up with this branched off of my changes from #209, so I'm going to wait until that is merged to get this one ready to merge.

**Comment by rsammelson** [comment]
This is now waiting on #470 instead.

**Comment by rsammelson** [comment]
@aeharding can you take a look at this?

File: src/helpers/vote.ts [hunk 1]
```diff
@@ -7,5 +7,19 @@ export function calculateCurrentVotesCount(
...
+  const upvotes =
+    item.counts.upvotes -
+    (item.my_vote === 1 ? 1 : 0) +
+    (votesById[id] === 1 ? 1 : 0);
+  const downvotes =
+    item.counts.downvotes -
+    (item.my_vote === -1 ? 1 : 0) +
+    (votesById[id] === -1 ? 1 : 0);
+
+  return {
```
Comment (aeharding) [inline src/helpers/vote.ts:20]: This function is called a lot, and separate upvote/downvote is not necessary for the majority of calls, as well as the happy path.

Can you split calculating separate upvotes/downvotes into a separate function and only call when necessary (when the setting is turned on for that)?

File: src/features/settings/appearance/Voting.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const voteDisplayMode = useAppSelector(
+    (state) => state.settings.appearance.voting.voteDisplayMode
+  );
+
+  const Selector = SettingSelector<VoteDisplayMode>;
+
+  return (
+    <>
+      <ListHeader>
+        <IonLabel>Voting</IonLabel>
```
Comment (aeharding) [inline src/features/settings/appearance/Voting.tsx:18]: Votes

File: src/features/settings/appearance/Voting.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      </ListHeader>
+      <IonList inset>
+        <Selector
+          title="Voting Display Mode"
```
Comment (aeharding) [inline src/features/settings/appearance/Voting.tsx:22]: How about

Display Votes - 

Cumulative

Separate

Hide

**Comment by aeharding** [comment]
@rsammelson I made some adjustment here, if you want to cherrypick or I can PR to yours.

https://github.com/aeharding/voyager/commit/aea3ff8e17b0ab2d8ef969f4a59b0f104b4793ab

**Comment by rsammelson** [comment]
I just saw your comment, I must have started working on it at the same time as you. I brought over some more changes from your commit in the most recent push.

**Comment by aeharding** [comment]
Oh oops!

Can we keep "cumulative"? I have that in e8861fbd41be170f8b834d0d21f6a2122f19a023 and I would add it but it looks like I don't have permission to push to your branch.

**Comment by rsammelson** [comment]
Yeah I can change it, I called it net though which seems more precise

**Comment by aeharding** [comment]
"Total" would also work. I want to leave off "Score" though because it's not consistent with the other options to add for that one only

Do you know if there's some config you can adjust so I may push to this branch?

**Comment by rsammelson** [comment]
Yeah it's a setting on my end but it says "access to secrets":


Which sounded scary so I didn't check it.

**Comment by rsammelson** [comment]
I adjusted it now

**Comment by aeharding** [comment]
Please check that option. I don't know why github words it that way.

**Comment by rsammelson** [comment]
Yeah on reading the help page the secrets can only possibly give modify access to other branches of the same repo, so it's not actually a security issue in this case

**Comment by rsammelson** [comment]
When I rebased on to main a while ago I forgot to apply the cleanup I did to the new settings.

**Comment by aeharding** [comment]
Thanks. This looks good. I wonder if the downvotes should be hidden if none in separate mode. In the comments it gets a little busy with a bunch of zeroes that are not necessary. that's something for another pr though

**Pull request merged**